### PR TITLE
Cleanup assets in build script ZIP file output.

### LINKED ISSUE #38446: [Woo AI] Create a `build` script for the plugin to package it as a ZIP file
We should be able to run `pnpm run build` and get a .ZIP file containing a fully-compatible WordPress.org plugin folder.

- [x] [Create a basic script](https://github.com/woocommerce/woocommerce/blob/add/woo-ai-plugin/plugins/woo-ai/bin/build-zip.sh) to build the ZIP plugin.
- [x] Verify that it's not missing anything required (you should be able to install it on a Jetpack-powered site).
- [x] Remove any extraneous dev files (.scss, *.ts, *.tsx, dev composer dependencies).
- [x] Create a changelog file.
- [x] Create a README file.
- [x] Make sure the main plugin file has plugin headers that satisfy WordPress.org plugin requirements.

Requirements:

>For Github repositories releases must be built and the zip files must be laid out like the dotorg repo generated zip files
Generally speaking this means inside your zipfile for your plugin named “myplugin” all of your repository files should be in a “myplugin” folder.

### PR DESCRIPTION
**Pull request by tommyshellberg**
### Changes proposed in this Pull Request:

- Update the Plugin URI in the main plugin file headers.
- Add a README.md file (included in ZIP file).
- Add a DEVELOPMENT.md file (not included in ZIP file).
- Generate and include a CHANGELOG.md file.
- Ignore some dev-specific files when building ZIP file.
- Ignore the /vendor/ directory when building ZIP file.
- Ignore the /woo-ai/ directory when building ZIP file.

Closes #38446.

### How to test the changes in this Pull Request:

1. Run `pnpm run build:zip`
2. Uncompress the ZIP file.
3. Test that it works properly via upload to an Atomic/self-hosted site.
4. Check that we aren't including any extra files.

### CODE REVIEW

File: plugins/woo-ai/DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Woo AI
+
+Woo AI is a WooCommerce plugin that utilizes the power of artificial intelligence to enhance your eCommerce experience. With features like AI-powered product title optimization and automated product description generation, Woo AI is designed to boost your store's efficiency and sales potential.
```
Comment (joelclimbsthings) [inline plugins/woo-ai/DEVELOPMENT.md:3]: 🥳

File: plugins/woo-ai/DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+pnpm -- wp-env start
+# Stop will, well, stop the environment
+pnpm -- wp-env stop
+# Destroy will remove all of the environment's files.
+pnpm -- wp-env destroy
+```
+
+## Development
+
+To enable Live(Hot) Reload when code is changed, run the following commands:
```
Comment (joelclimbsthings) [inline plugins/woo-ai/DEVELOPMENT.md:27]: We might mention the `pnpm run build` script here as well.

**Review by joelclimbsthings — APPROVED** [review]
Thanks for all the cleanup here @tommyshellberg 💯 . Stoked to see the *.MD files, and everything tested well when uploaded to the Atomic site. I had one minor optional comment.

File: plugins/woo-ai/DEVELOPMENT.md (hunk 2 above)
Comment (tommyshellberg) [reply plugins/woo-ai/DEVELOPMENT.md:27]: I wasn't sure how to describe its use case but I think this makes sense?

>To build the /woo-ai/ plugin directory (when loading the plugin via symlink), run:
```
pnpm install
pnpm run build
```

Looking over this helped me catch a typo with the command too: 😆 

```text
pnpm run pnpm run build
```

**Pull request merged**